packages/adapter-node/src/LoadNodeTexture.ts

```diff
@@ -0,0 +1,38 @@
+import { Texture } from '@pixi/core';
+import { getResolutionOfUrl } from '@pixi/utils';
+import { CanvasRenderingContext2D, loadImage } from 'canvas';
+import { NodeCanvasElement } from './NodeCanvasElement';
+
+const validImages = ['jpg', 'png', 'jpeg'];
+
+/**
+ * loads our textures!
+ * this makes use of imageBitmaps where available.
+ * We load the ImageBitmap on a different thread using CentralDispatch
+ * We can then use the ImageBitmap as a source for a Pixi Texture
+ */
+export const loadNodeTexture = {
+    test(url: string): boolean
+    {
+        const tempURL = url.split('?')[0];
+        const extension = tempURL.split('.').pop();
```

Comment:
You should use the Node URL object here to parse the extension. It'll be much more accurate and less prone to issues parsing.

```js
import { URL } from 'url';
import * as path from 'path';

const url = new URL(url);
const extension = path.extension(url.pathname);
```